realm/realm-library/src/androidTest/java/io/realm/RealmAnnotationTests.java

```diff
@@ -127,10 +127,10 @@ public void primaryKey_errorOnInsertingSameObject() {
     }
 
     @Test
-    public void primaryKey_isIndexed() {
+    public void string_primaryKey_isNotIndexed() {
         Table table = realm.getTable(PrimaryKeyAsString.class);
         assertNotNull(OsObjectStore.getPrimaryKeyForObject(realm.getSharedRealm(), PrimaryKeyAsString.CLASS_NAME));
-        assertTrue(table.hasSearchIndex(table.getColumnKey("name")));
+        assertFalse(table.hasSearchIndex(table.getColumnKey("name")));
```

Comment:
Primary keys are not indexed by default anymore? :thinking:

Reply:
With Core 6 we are mapping user defined String primary keys to our internal ObjectId which is faster so the index isn't strictly needed for looking up single objects.

That said, it is possible that we are forgetting about some use cases or that we are missing some upgrade tests. Thanks for pointing it out 👏

Reply:
There is no migration needed, there's a test covering this use case in https://github.com/realm/realm-java/pull/6107/commits/d113014e9506a49e41fcd290df2d10e634aa7c6f#diff-fe5ca9f6e0254b639a10bcc000406fdbR1439